core/dbt/events/types.py

```diff
@@ -299,6 +300,295 @@ def cli_msg(self) -> str:
+    file_dict: Dict
+
+    def cli_msg(self) -> str:
+        return f"PP file: {self.file_dict}"
+
+
+@dataclass
+class PartialParsingException(DebugLevel, CliEventABC):
+    exc_info: Dict
+
+    def cli_msg(self) -> str:
+        return f"PP exception info: {self.exc_info}"
+
+
+class PartialParsingSkipParsing(DebugLevel, CliEventABC):
+    def cli_msg(self) -> str:
+        return "Partial parsing enabled, no changes found, skipping parsing"
+
+
+class PartialParsingMacroChangeStartFullParse(InfoLevel, CliEventABC):
+    def cli_msg(self) -> str:
+        return "Change detected to override macro used during parsing. Starting full parse."
+
+
+@dataclass
+class ManifestWrongMetadataVersion(DebugLevel, CliEventABC):
+    version: str
+
+    def cli_msg(self) -> str:
+        return ("Manifest metadata did not contain correct version. "
+                f"Contained '{self.version}' instead.")
+
+
+@dataclass
+class PartialParsingVersionMismatch(InfoLevel, CliEventABC):
+    saved_version: str
+    current_version: str
+
+    def cli_msg(self) -> str:
+        return ("Unable to do partial parsing because of a dbt version mismatch. "
+                f"Saved manifest version: {self.saved_version}. "
+                f"Current version: {self.current_version}.")
+
+
+class PartialParsingFailedBecauseConfigChange(InfoLevel, CliEventABC):
+    def cli_msg(self) -> str:
+        return ("Unable to do partial parsing because config vars, "
+                "config profile, or config target have changed")
+
+
+class PartialParsingFailedBecauseProfileChange(InfoLevel, CliEventABC):
+    def cli_msg(self) -> str:
+        return ("Unable to do partial parsing because profile has changed")
+
+
+class PartialParsingFailedBecauseNewProjectDependency(InfoLevel, CliEventABC):
+    def cli_msg(self) -> str:
+        return ("Unable to do partial parsing because a project dependency has been added")
+
+
+class PartialParsingFailedBecauseHashChanged(InfoLevel, CliEventABC):
+    def cli_msg(self) -> str:
+        return ("Unable to do partial parsing because a project config has changed")
+
+
+class PartialParsingNotEnabled(DebugLevel, CliEventABC):
+    def cli_msg(self) -> str:
+        return ("Partial parsing not enabled")
+
+
+# TODO: add in exc_info=True
+@dataclass
+class ParsedFileLoadFailed(DebugLevel, CliEventABC):
+    path: str
+    exc: Exception
+
+    def cli_msg(self) -> str:
+        return f"Failed to load parsed file from disk at {self.path}: {self.exc}"
+
+
+class PartialParseSaveFileNotFound(InfoLevel, CliEventABC):
+    def cli_msg(self) -> str:
+        return ("Partial parse save file not found. Starting full parse.")
+
+
+@dataclass
+class StaticParserCausedJinjaRendering(DebugLevel, CliEventABC):
+    path: str
+
+    def cli_msg(self) -> str:
+        return f"1605: jinja rendering because of STATIC_PARSER flag. file: {self.path}"
+
+
+@dataclass
+class UsingExperimentalParser(DebugLevel, CliEventABC):
```

Comment:
Can you throw a TODO in here for us marking some of these as `TestLevel`? The experimental parser / static parser code uses these for testing, but we might want to leave some as debug. A decision to make later though.